### DISCUSSION TITLE
Making environment variable defined in .yml workflow available during JVM runtime

### DISCUSSION THREAD

**Discussion by mazgajmaciek** [actions/virtual-environments#5126] [General questions]
I have an android app that generates separate builds for each environment we have available - due to that we also have separate test users for each environment (e.g. dev-john, stg-john etc.) so I need to have a way to modify which user is being used for corresponding environment.

I have tried adding env variable in .yml workflow as such:
`env:
    ENVIRONMENT: "dev"`

So then based on that variable I would like to retrieve it via something like `System.getProperty()` and select the appropriate user for the environment. Is this doable? When doing `printenv` I can see the ENVIRONMENT variable is getting propagated in GH Actions build log however it only gives me `null` when I try to retrieve it in test class with System.getProperty("ENVIRONMENT") during JVN runtime. I also tried using `System.getenv("ENVIRONMENT")` since I'm using macos runner but still getting only `null` at runtime.

**Comment by nikolai-frolov** [comment]
Please see GitHib Workflow to write correct yml ([example](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-5)). Pay attention that yml is sensitive to line breaks and white spaces.
